gitbrute.go

```diff
@@ -79,7 +79,7 @@ func main() {
 	close(done)
 
 	cmd := exec.Command("git", "commit", "--allow-empty", "--amend", "--date="+w.author.String(), "--file=-")
-	cmd.Env = append([]string{"GIT_COMMITTER_DATE=" + w.committer.String()}, os.Environ()...)
+	cmd.Env = append(append([]string(nil), os.Environ()...), "GIT_COMMITTER_DATE="+w.committer.String())
```

Comment:
why the nil in there?

You can just `cmd.Env = append(os.Environ(), "GIT_COMMITTER_DATE="+w.committer.String())`

If the concern was aliasing, you can do:

```go
  env := os.Environ()
  cmd.Env = append(env[:len(env):len(env], "GIT_COMMITTER_DATE="+w.committer.String())
```

Reply:
BTW, some history: this gitbrute code predates the change I made in Go to define the environment variable order meaning:

https://golang.org/doc/go1.9#os/exec (in 2017: https://github.com/golang/go/commit/e73f4894949c4ced611881329ff8f37805152585)

Reply:
You guessed exactly right, I didn't know that `os.Environ()` is guaranteed to return an independent copy of the environment and passes ownership to the caller, I should have read its documentation.

Regarding the `slice[a:b:c]` expressions, I didn't know that these are guaranteed to always creates a new slice _plus_ a new underlying array, [the documentation](https://golang.org/ref/spec#Slice_expressions) doesn't mention this as clearly as I would expect. Thanks for telling me, I'll remember it for sure now.

Reply:
A slice-with-capacity doesn't make a new slice: but append necessarily makes a copy of it. (Otherwise there's no place to put the appended values)

Reply:
Thanks, after some more experiments I came to the same conclusion. Quite tricky, but consistent and simple. 👍